WooCommerce/Classes/ViewRelated/Dashboard/Orders/LastOrdersDashboardCardViewModel.swift

```diff
@@ -0,0 +1,199 @@
+
+    init(siteID: Int64,
+         stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.siteID = siteID
+        self.analytics = analytics
+        self.stores = stores
+        self.storageManager = storageManager
+    }
+
+    @MainActor
+    func reloadData() async {
+        syncingData = true
+        syncingError = nil
+        rows = []
+        configureResultsController()
+
+        do {
+            // Send network request -> listen to storage -> load UI
+            try await loadLast3Orders(for: selectedOrderStatus)
+        } catch {
+            syncingError = error
+            DDLogError("⛔️ Dashboard (Last orders) — Error loading orders: \(error)")
+        }
+        syncingData = false
+    }
+
+    func dismiss() {
+        analytics.track(event: .DynamicDashboard.hideCardTapped(type: .lastOrders))
+        onDismiss?()
+    }
+
+    func updateOrderStatus(_ status: OrderStatusRow) {
+        selectedOrderStatus = status.status
+
+        Task {
+            await reloadData()
+        }
+    }
+}
+
+private extension LastOrdersDashboardCardViewModel {
+    func ordersPredicate() -> NSPredicate {
+        let sitePredicate = NSPredicate(format: "siteID == %lld", siteID)
+
+        guard let slug = selectedOrderStatus?.rawValue else {
+            return sitePredicate
+        }
+
+        let statusPredicate = NSPredicate(format: "statusKey == [c] %@", slug)
+        return NSCompoundPredicate(andPredicateWithSubpredicates: [statusPredicate, sitePredicate])
+    }
+
+    @MainActor
+    func loadLast3Orders(for status: OrderStatusEnum?) async throws {
+        return try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(OrderAction.synchronizeOrders(
+                siteID: siteID,
+                statuses: [status?.rawValue].compactMap { $0 },
+                pageNumber: Constants.pageNumber,
+                pageSize: Constants.numberOfOrdersToShow,
+                onCompletion: { _, error in
+                    if let error {
+                        continuation.resume(throwing: error)
+                    } else {
+                        continuation.resume()
+                    }
+                }))
+        }
+    }
+
+    func configureResultsController() {
+        let sortDescriptorByID = NSSortDescriptor(keyPath: \StorageOrder.dateCreated, ascending: false)
+        let resultsController = ResultsController<StorageOrder>(storageManager: storageManager,
+                                                                matching: ordersPredicate(),
+                                                                fetchLimit: Constants.numberOfOrdersToShow,
+                                                                sortedBy: [sortDescriptorByID])
+        self.resultsController = resultsController
```

Comment:
Interesting that it's creating a new `ResultsController` for every reload instead of updating the predicate like I did in Reviews card. This does make the code easier to follow imo 👍🏼